File: apps/beeai-sdk/src/beeai_sdk/platform/context.py
# Copyright 2025 © BeeAI a Series of LF Projects, LLC
# SPDX-License-Identifier: Apache-2.0

import os

import httpx

from beeai_sdk.util import resource_context

get_client, with_client = resource_context(
    httpx.AsyncClient, lambda: httpx.AsyncClient(base_url=os.environ.get("PLATFORM_URL", "http://127.0.0.1:8333"))
)

__all__ = [
    "get_client",
    "with_client",
]

File: apps/agentstack-sdk-ts/src/experimental/server/context.ts
/**
 * Copyright 2025 © BeeAI a Series of LF Projects, LLC
 * SPDX-License-Identifier: Apache-2.0
 */

import type { Task } from '@a2a-js/sdk';

export class RunContext {
  public readonly taskId: string;
  public readonly contextId: string;
  public readonly task?: Task;

  constructor(taskId: string, contextId: string, task?: Task) {
    this.taskId = taskId;
    this.contextId = contextId;
    this.task = task;
  }
}
